docs/docs/module_guides/workflow/index.md

```diff
@@ -632,6 +632,54 @@ result = await handler
 handler = w.run(ctx=handler.ctx)
 result = await handler
 ```
+## Resources
+
+Resources are external dependencies that you can equip the steps of your workflows with.
```

Comment:
I'd use something like "Resources are external dependencies you can inject into steps of a workflow" to convey the "dependency injection" concept.